File: mall-admin/src/api/operationLog.ts
import { request } from '@/utils/request'
import type { PageResult } from '@/types'

export interface OperationLog {
  id: number
  adminId?: number
  adminName?: string
  module?: string
  operation?: string
  method?: string
  params?: string
  ip?: string
  status?: number
  costTime?: number
  createTime?: string
  eventType?: string
  userCouponId?: number
  userId?: number
  couponId?: number
  orderId?: number
  remark?: string
}

export function getOperationLogPage(params: { current: number; size: number; status?: number; eventType?: string; userId?: number }) {
  return request<PageResult<OperationLog>>({ url: '/operation-logs/page', method: 'get', params })
}

File: mall-user/src/main/java/com/mall/entity/OperationLog.java
package com.mall.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("operation_log")
public class OperationLog implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long adminId;

    private String adminName;

    private String module;

    private String operation;

    private String method;

    private String params;

    private String ip;

    private Integer status;

    private Long costTime;

    private String eventType;
    private Long userCouponId;
    private Long userId;
    private Long couponId;
    private Long orderId;
    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}
